email.go

```diff
@@ -420,6 +423,32 @@ func (email *Email) SetSubject(subject string) *Email {
 	return email
 }
 
+// SetListUnsubscribe sets the Unsubscribe address.
+func (email *Email) SetListUnsubscribe(address string) *Email {
+	if email.Error != nil {
+		return email
+	}
+
+	email.AddHeader("List-Unsubscribe", address)
+
+	return email
+}
+
+// SetDkim adds DomainKey signature to the email message (header+body)
+func (email *Email) SetDkim(options dkim.SigOptions) *Email {
+
+	msg := []byte(email.GetMessage())
+	err := dkim.Sign(&msg, options)
+
+	if err != nil {
+		return email
+	}
+
+	email.DkimMsg = string(msg)
+
+	return email
+}
+
```

Comment:
Can you replace function with this to return error early and set error to email.Error please?
```go
// SetDkim adds DomainKey signature to the email message (header+body)
func (email *Email) SetDkim(options dkim.SigOptions) *Email {
	if email.Error != nil {
		return email.Error
	}

	msg := []byte(email.GetMessage())
	err := dkim.Sign(&msg, options)
	if err != nil {
		email.Error = errors.New("Mail Error: cannot dkim sign message due: %s" + err.Error())
		return email
	}

	email.DkimMsg = string(msg)

	return email
}
```